content/en/blog/2024/target-allocator-troubleshooting/index.md

```diff
@@ -0,0 +1,481 @@
+
+```yaml
+apiVersion: opentelemetry.io/v1beta1
+kind: OpenTelemetryCollector
+metadata:
+  name: otelcol
+  namespace: opentelemetry
+spec:
+  mode: statefulset
+  targetAllocator:
+    enabled: true
+    serviceAccount: opentelemetry-targetallocator-sa
+    prometheusCR:
+      enabled: true
+```
+
+📝 You can see the full `OpenTelemetryCollector` resource definition
+[here](https://github.com/avillela/otel-target-allocator-talk/blob/main/src/resources/02-otel-collector.yml).
+
+### 4- Did you configure a ServiceMonitor (or PodMonitor) selector?
+
+If you configured a
+[`ServiceMonitor`](https://observability.thomasriley.co.uk/prometheus/configuring-prometheus/using-service-monitors/#:~:text=The%20ServiceMonitor%20is%20used%20to,build%20the%20required%20Prometheus%20configuration.)
+selector, it means that the Target Allocator will only look for
+`ServiceMonitors` having a `metadata.label` that matches the value in
+[`serviceMonitorSelector`](https://github.com/open-telemetry/opentelemetry-operator/blob/main/docs/api.md#opentelemetrycollectorspectargetallocatorprometheuscr).
+
+Suppose that you configured a
+[`serviceMonitorSelector`](https://github.com/open-telemetry/opentelemetry-operator/blob/main/docs/api.md#opentelemetrycollectorspectargetallocatorprometheuscr)
+for your Target Allocator, like in the example below:
+
+```yaml
+apiVersion: opentelemetry.io/v1beta1
+kind: OpenTelemetryCollector
+metadata:
+  name: otelcol
+  namespace: opentelemetry
+spec:
+  mode: statefulset
+  targetAllocator:
+    enabled: true
+    serviceAccount: opentelemetry-targetallocator-sa
+    prometheusCR:
+      enabled: true
+      serviceMonitorSelector:
+        app: my-app
+```
+
+It means that your `ServiceMonitor` resource must in turn have that same value
+in `metadata.spec.label`:
+
+```yaml
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: sm-example
+  labels:
+    app: my-app
+    release: prometheus
+spec:
+```
+
+📝 You can see the full `ServiceMonitor` resource definition
+[here](https://github.com/avillela/otel-target-allocator-talk/blob/main/src/resources/04-service-monitor.yml).
+
+In this case, the `OpenTelemetryCollector` resource's
+`prometheusCR.serviceMonitorSelector` is looking only for `ServiceMonitors`
+having the label `app: my-app`, which we see in the above example.
+
+**_If your ServiceMonitor resource is missing that label, then the Target
+Allocator won’t pick up that ServiceMonitor._**
+
+> **NOTE:** The same applies if you’re using a
+> [PodMonitor](https://prometheus-operator.dev/docs/user-guides/getting-started/#using-podmonitors).
+> In that case, if your `OpenTelemetryCollector`resource defined a
+> [`podMonitorSelector`](https://github.com/open-telemetry/opentelemetry-operator/blob/main/docs/api.md#opentelemetrycollectorspectargetallocatorprometheuscr),
+> then any `PodMonitors` you wish to be picked up by the TargetAllocator would
+> need to have that same label.
```

Comment:
Not sure this needs to be a call-out since you address other PodMonitor variations in the text, but I'm fine either way. If keeping the call-out, consider switching it to the docsy shortcode (for example, [this tip](https://github.com/open-telemetry/opentelemetry.io/blob/e7cbabb5fdf17b7e7355ec6723f69631d61400b3/content/en/docs/contributing/new-content.md?plain=1#L15)).